firebase-perf/src/main/java/com/google/firebase/perf/util/Rate.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf.util;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.concurrent.TimeUnit;
+
+/** A Rate object representing the number of tokens per total specified time unit. */
+public class Rate {
+
+  private long numTokensPerTotalTimeUnit;
+  private long numTimeUnits;
+  private TimeUnit timeUnit;
+
+  /**
+   * Constructs a Rate object.
+   *
+   * <p>For example, a rate of 3 tokens per minute can be represented by new Rate(3, 1, MINUTES) or
+   * new Rate(3, 60, SECONDS).
+   *
+   * @param numTokensPerTotalTimeUnit The number of tokens to be issued within the specified time
+   * @param numTimeUnits The number of specified time unit
+   * @param timeUnit The specified time unit
+   */
+  public Rate(long numTokensPerTotalTimeUnit, long numTimeUnits, TimeUnit timeUnit) {
```

Comment:
This is to prevent bugs like the existing one; if you only pass an int or long, it's very easy to get the units wrong (before we were mixing up minutes and seconds). By passing this, it's impossible to get it wrong.

Reply:
I think the third parameter is there mainly for readability reasons. Before in this class, there are two conflicting comments about what the rate is: one says it's token per minute and the other one says it's token per second. By explicitly passing in the TimeUnit, we make it very clear that new Rate(3, 60, SECONDS) means 3 tokens per 60 seconds, and not 3 tokens per 60 minutes or vice versa.

Besides, by keeping the third parameter, we can also make testing more readable, by specifying rates of units other than SECONDS. For example, we can specify new Rate(2, 1, MINUTE) for 2 tokens per minute.

Reply:
I definitely see the need for the readability part. My suggestion was to create the readability by calling the time unit to be always in seconds. Eg. (numberOfTokensForTimeLimit, timeLimitInSeconds)

The current is definitely super verbose and helpful. Was just feeling that verbosity can come by qualifying the parameter with better names.

Anyways, not too tied to my thoughts on this. Fine with this change as well.